File: src/my_project_typescript/easy_problems_typescript/from101to150/fairCandySwap.ts
function fairCandySwap(aliceSizes: number[], bobSizes: number[]): number[] {

    let sumOne: number = aliceSizes.reduce((a,b) => a+b,0)
    let sumTwo: number = bobSizes.reduce((a,b) => a+b,0)
    let gap: number = 0
    if (sumOne > sumTwo){
        gap = Math.floor((sumOne-sumTwo)/2)
        for (let i = 0; i < aliceSizes.length;i++){
            if (bobSizes.includes( aliceSizes[i] - gap)){
                return [aliceSizes[i],aliceSizes[i]-gap]
            }
        }
    }
    else {
        gap = Math.floor((sumTwo-sumOne)/2)
        for (let i = 0; i < bobSizes.length;i++){
            if (aliceSizes.includes( bobSizes[i] - gap)){
                return [ bobSizes[i]-gap, bobSizes[i]]
            }
        }
    }
    return []
};


console.log(fairCandySwap([1,1], [2,2]))

console.log(fairCandySwap([1,2],  [2,3]))

console.log(fairCandySwap([2], [1,3]))

console.log(fairCandySwap([1,2,5], [2,4]))

File: src/my_project/easy_problems/from201to250/fair_candy_swap.py
from typing import List 

class Solution:
    def fairCandySwap(self, aliceSizes: List[int], bobSizes: List[int]) -> List[int]:
        sum1 = sum(aliceSizes)
        
        sum2 = sum(bobSizes)
        
        if sum1 > sum2:
            gap = (sum1 - sum2)//2
            for i in aliceSizes:
                if i-gap in bobSizes:
                    return [i,i-gap]
        else:
            gap = (sum2 - sum1)//2
            for i in bobSizes :
                if i-gap in aliceSizes:
                    return [i-gap,i]
                
solution = Solution()

print(solution.fairCandySwap(aliceSizes = [1,1], bobSizes = [2,2]))

print(solution.fairCandySwap(aliceSizes = [1,2], bobSizes = [2,3]))

print(solution.fairCandySwap(aliceSizes = [2], bobSizes = [1,3]))

print(solution.fairCandySwap(aliceSizes = [1,2,5], bobSizes = [2,4]))


